com.unity.formats.fbx/Editor/ExportModelSettings.cs

```diff
@@ -164,8 +188,25 @@ internal abstract class ExportOptionsSettingsSerializeBase : IExportOptions
         public abstract ExportSettings.LODExportType LODExportType { get; }
         public abstract ExportSettings.ObjectPosition ObjectPosition { get; }
         public abstract bool ExportUnrendered { get; }
-        public virtual bool PreserveImportSettings { get;  }
+        public virtual bool PreserveImportSettings { get { return false; } }
         public abstract bool AllowSceneModification { get; }
+
+        public override bool Equals(object e)
+        {
+            var expOptions = e as ExportOptionsSettingsSerializeBase;
+            if (expOptions == null)
+            {
+                return false;
+            }
+            return animatedSkinnedMesh == expOptions.animatedSkinnedMesh &&
+                mayaCompatibleNaming == expOptions.mayaCompatibleNaming &&
+                exportFormat == expOptions.exportFormat;
+        }
+
+        public override int GetHashCode()
+        {
+            return (animatedSkinnedMesh ? 1 : 0) | (mayaCompatibleNaming ? 1 : 0) << 1 | (int)exportFormat << 2;
```

Comment:
can you add parens around the bitshifts? I want to make clear that you're shifting `(mayaCompatibleNaming ? 1 : 0)` and not `(animatedSkinnedMesh ? 1 : 0) | (mayaCompatibleNaming ? 1 : 0)`